Gh enterprise 994

danantonioreyes opened the pull request.

bendk commented on apps/videos/forms.py:
```diff
@@ -413,6 +415,32 @@ def needs_primary_audio_language(self):
+    primary_audio_language_code = AmaraChoiceField(
+                label=_('This video is in:'),
+                help_text=_('Please double check the primary spoken '
+                            'language. This step cannot be undone.'))
```
I think this help text could use some work:

 - Maybe something like "Video Language" for the label.  I think that's what we use in other places
 - The help text isn't correct, for team videos admins/managers can change the primary audio language through the edit video form.

danantonioreyes replied:
Changed the label to "Video language" to make it more consistent with the label we use in futureui.
Removed the help text since it was already being set in the form's `__init__()` method

bendk commented on templates/future/videos/create-subtitles-modal.html:
```diff
@@ -0,0 +1,32 @@
+{% load i18n forms %}
```
I think you need to use the new modal code for this one.

bendk commented on apps/videos/forms.py:
```diff
@@ -413,6 +415,32 @@ def needs_primary_audio_language(self):
+            if can_edit_videos(team, request.user):
+                self.fields['primary_audio_language_code'].help_text = ''
+            else:
+                self.fields['primary_audio_language_code'].help_text = 'Please double check the primary spoken language.'
```
I'm not sure why the help text would be different depending on can_edit_videos().  I personally think the help text can always be blank.

bendk replied:
That makes sense, I'm just wondering if the help text is useful though.  How can you double check the language at this point?  What are you supposed to do if the language is incorrect?

My feeling is no help text is better.  @kslottow @PCF-Testing what do you think

bendk commented on apps/subtitles/workflows.py:
```diff
@@ -358,6 +358,14 @@ def user_can_edit_video(self, user):
         """
         raise NotImplementedError()
 
+    def user_can_subtitle_video(self, user):
```
Maybe it should be something like `user_can_create_new_subtitles` to distinguish it more from `user_can_edit_subtitles`.  I also think some words in the docstring about the difference between these 2 methods could go a long way.

bendk commented on apps/subtitles/workflows.py:
```diff
@@ -358,6 +358,14 @@ def user_can_edit_video(self, user):
+        Returns:
+            True/False
+        """
+        raise NotImplementedError()
```
If you raise NotImplementedError, will this fail for simple/collab teams?  Maybe it's better to return some default.

danantonioreyes replied:
The default is implemented in `DefaultVideoWorkflow`

danantonioreyes replied:
the default is in `DefaultVideoWorkflow`.

For collab teams, the inheritance goes: `CollaborationWorkflow` --> `TeamVideoWorkflow` --> `DefaultVideoWorkflow`

danantonioreyes commented:
Thanks for the review @bendk ! I have addressed some of the review items and have commented on the others. Will wait for the deploy before merging this to dev

PCF-Testing commented:
> That makes sense, I'm just wondering if the help text is useful though. How can you double check the language at this point? What are you supposed to do if the language is incorrect?

> My feeling is no help text is better.

Double-checking would involve pressing Cancel, playing the video in the embedder and making sure the what the language of the video is.

If the language is set incorrectly, a member of a simple team would have to contact a team admin or Amara support for help, and a public user would have to ask support for assistance.

As a user, I'd prefer to know in advance if some data I am about to enter cannot be changed later, so I am for keeping the help text.

bendk commented:
I still don't like it because I doubt any user is actually going to go through with those steps, nor do I think it's really reasonable to ask them to do that.

That said, this is only going to come up in fringe cases, so I'm fine with keeping the text for now and working on it later.

BTW, on the subject of future improvements.  Why does the video language need to be set in this case?  Maybe we can just leave that out and let team admins set it.

PCF-Testing commented:
> Why does the video language need to be set in this case? Maybe we can just leave that out and let team admins set it.

I have thought about this. For public videos and videos within simple teams it is not a critical piece of info, rather something that is needed for the convenience of the subtitle makers and the audience. However, the primary language becomes important if a video is moved to a task- or collab-based team with a rigorous workflow. These teams assume that a subtitled video has the primary language known and set.
